src/sentry/api/endpoints/project_rule_actions.py

```diff
@@ -0,0 +1,59 @@
+from rest_framework.exceptions import ValidationError
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import ProjectAlertRulePermission, ProjectEndpoint
+from sentry.api.serializers.rest_framework import RuleActionSerializer
+from sentry.models import Rule
+from sentry.rules.processor import RuleProcessor
+from sentry.utils.safe import safe_execute
+from sentry.utils.samples import create_sample_event
+from sentry.web.decorators import transaction_start
+
+
+@region_silo_endpoint
+class ProjectRuleActionsEndpoint(ProjectEndpoint):
+    private = True
+    permission_classes = (ProjectAlertRulePermission,)
+
+    @transaction_start("ProjectRuleActionsEndpoint")
+    def post(self, request: Request, project) -> Response:
+        """
+        Activate a rule's downstream actions using the data from a previous event
+
+            {method} {path}
+            {{
+                "actions": []
+            }}
+
+        """
+        serializer = RuleActionSerializer(
+            context={"project": project, "organization": project.organization}, data=request.data
+        )
+
+        if not serializer.is_valid():
+            raise ValidationError
+
+        data = serializer.validated_data
+        if len(data.get("actions", [])) == 0:
+            raise ValidationError
+
+        data["conditions"] = []
+        data["filters"] = []
+        data["actionMatch"] = "all"
+        data["filterMatch"] = "all"
+        data["frequency"] = 30
+        rule = Rule(project=project, data=data)
+
+        test_event = create_sample_event(
```

Comment:
This is the same method used in `ProjectCreateSampleEndpoint`. This event isn't deleted at the end of the method because I think some of the async notification tasks require it? Sending a notification to a user gave me `sentry.tasks.digests: Skipped digest delivery due to empty digest`

Reply:
This should work ok. events created this way skip post processing and so we don't double send them or anything like that.

I'm not sure why the digest is being skipped. You might need to debug through https://github.com/getsentry/sentry/blob/fe07466a1449a5ae60526528ce7bf9399b59b47d/src/sentry/mail/adapter.py#L67-L81 and see where it's being dropped.

Reply:
This might be part of your problem https://github.com/getsentry/sentry/blob/fe07466a1449a5ae60526528ce7bf9399b59b47d/src/sentry/digests/notifications.py#L44-L52

Since you're creating a fake `Rule` it might not work as expected. Could be worth trying it with a real rule and seeing if it just works.

It might be painful to get emails to work without a real rule, but we can probably figure something out.

Reply:
digest was only being skipped if I deleted the rule, doing it this way doesn't skip it

Reply:
Are notification emails supposed to work locally? From the logger I can see that the test notification gets digested, but the scheduled task sends the emails doesn't seem to be running

Reply:
We don't have an email server running by default, so emails won't send out. There is `sentry run smtp` although I haven't used it, so not sure how it works and what else you might need to set up. https://github.com/getsentry/sentry/blob/c16c78edfae643f22b490d64fe46cac524eacdf5/src/sentry/runner/commands/run.py#L119-L144

How are you testing locally? If you're running `sentry devserver`, you'll need to use `--workers` to make sure celery is running. I think you might also need to run celerybeat, `sentry run cron` should do it https://github.com/getsentry/sentry/blob/c16c78edfae643f22b490d64fe46cac524eacdf5/src/sentry/runner/commands/run.py#L274-L302

Reply:
thanks, `sentry run cron` did it! and also just tested saving the rule and it worked, otherwise all the items are reduced on the `group_records` step of the pipeline

Reply:
What's behind the logic for `immediate_delivery`? Sometimes the notification is digested immediately and sometimes it isn't and disappears (the next scheduled tasks don't pick it up either)

Reply:
There's some magic that happens to decide whether a notification is digested or not. I don't actually know the details here though. But what happens is that it will eventually be sent in the next <time_period> depending on the digest settings of the project. https://sentry.io/settings/sentry/projects/sentry/alerts/

Reply:
This logic lives inside of a lua script in redis https://github.com/getsentry/sentry/blob/27ac186f03d07d17363d3e98e5555ef59503ac90/src/sentry/digests/backends/redis.py#L102-L140 :D

Reply:
do you think its okay to reserve/associate None or -1 as the id of a dummy rule? The only part that doesn't work right now is in the `rewrite_record` step of the pipeline where we try to look up the rule object using the id